src/stirling/source_connectors/socket_tracer/socket_trace_bpf_test.cc

```diff
@@ -747,6 +750,163 @@ TEST_F(NullRemoteAddrTest, IPv6Accept4WithNullRemoteAddr) {
+
+  bool found_port = false;
+  uint16_t port = ntohs(client_sockaddr.sin_port);
+  for (auto lport : GetLocalPorts(records, kHTTPLocalPortIdx, indices)) {
+    // TODO(ddelnano): Determine why the local_addr column is storing the port
+    // in network byte order.
+    LOG(INFO) << "Local port: " << lport << " and pre ntohs " << client_sockaddr.sin_port
+              << " and ntohs " << port;
+    if (lport == client_sockaddr.sin_port) {
+      found_port = true;
+      break;
+    }
+  }
+  EXPECT_TRUE(found_port);
+}
+
+TEST_F(LocalAddrTest, IPv6ConnectPopulatesLocalAddr) {
+  StartTransferDataThread();
+
+  TCPSocket client(AF_INET6);
+  TCPSocket server(AF_INET6);
+
+  std::atomic<bool> server_ready = false;
+
+  std::thread server_thread([&server, &server_ready]() {
+    server.BindAndListen();
+    server_ready = true;
+    auto conn = server.Accept(/* populate_remote_addr */ false);
+
+    std::string data;
+
+    conn->Read(&data);
+    conn->Write(kHTTPRespMsg1);
+  });
+
+  while (!server_ready) {
+  }
+
+  std::thread client_thread([&client, &server]() {
+    client.Connect(server);
+
+    std::string data;
+
+    client.Write(kHTTPReqMsg1);
+    client.Read(&data);
+  });
+
+  server_thread.join();
+  client_thread.join();
+
+  // Get the remote port seen by server from client's local port.
+  struct sockaddr_in6 client_sockaddr = {};
+  socklen_t client_sockaddr_len = sizeof(client_sockaddr);
+  struct sockaddr* client_sockaddr_ptr = reinterpret_cast<struct sockaddr*>(&client_sockaddr);
+  ASSERT_EQ(getsockname(client.sockfd(), client_sockaddr_ptr, &client_sockaddr_len), 0);
+
+  // Close after getting the sockaddr from fd, otherwise getsockname() wont work.
+  client.Close();
+  server.Close();
+
+  StopTransferDataThread();
+
+  std::vector<TaggedRecordBatch> tablets = ConsumeRecords(kHTTPTableNum);
+  ASSERT_NOT_EMPTY_AND_GET_RECORDS(const types::ColumnWrapperRecordBatch& record_batch, tablets);
+
+  std::vector<size_t> indices =
+      testing::FindRecordIdxMatchesPID(record_batch, kHTTPUPIDIdx, getpid());
+  ColumnWrapperRecordBatch records = testing::SelectRecordBatchRows(record_batch, indices);
+
+  ASSERT_THAT(records, RecordBatchSizeIs(2));
+
+  // Make sure that the socket info resolution works.
+  ASSERT_OK_AND_ASSIGN(std::string remote_addr, IPv6AddrToString(client_sockaddr.sin6_addr));
+  EXPECT_THAT(GetLocalAddrs(records, kHTTPLocalAddrIdx, indices), Contains("::1").Times(2));
+  EXPECT_EQ(remote_addr, "::1");
+
+  bool found_port = false;
+  uint16_t port = ntohs(client_sockaddr.sin6_port);
+  for (auto lport : GetLocalPorts(records, kHTTPLocalPortIdx, indices)) {
+    // TODO(ddelnano): Determine why the local_addr column is storing the port
+    // in network byte order.
+    LOG(INFO) << "Local port: " << lport << " and pre ntohs " << client_sockaddr.sin6_port
+              << " and ntohs " << port;
+    if (lport == client_sockaddr.sin6_port) {
+      found_port = true;
+      break;
+    }
```

Comment:
My suspicion for why this is happening is that we are calling `ntohs` more than once and so we end up reversing the endianess back to network byte order. I'm still in the process of tracking it down.

Reply:
The kernel's skc_num (also referred to as [sk_num](https://elixir.bootlin.com/linux/v6.10.7/source/include/net/sock.h#L359)) is stored in host byte order. I verified this through `bpf_trace_printk` calls and this example from the kernel [source](https://elixir.bootlin.com/linux/v6.10.7/source/net/ipv4/inet_diag.c#L79).